kotlin/workflow-tracing/src/main/java/com/squareup/workflow/diagnostic/tracing/TracingDiagnosticListener.kt

```diff
@@ -72,31 +72,53 @@ fun TracingDiagnosticListener(
   encoderProvider: (workflowScope: CoroutineScope) -> TraceEncoder
 ): TracingDiagnosticListener =
   TracingDiagnosticListener(memoryStats = memoryStats) { workflowScope, type ->
-    val encoder = encoderProvider(workflowScope)
-    val processName = name.ifEmpty { type }
-    encoder.createLogger(
-        processName = processName,
-        threadName = "Profiling"
-    )
+    provideLogger(name, workflowScope, type, encoderProvider)
   }
 
+internal fun provideLogger(
+  name: String,
+  workflowScope: CoroutineScope,
+  workflowType: String,
+  encoderProvider: (workflowScope: CoroutineScope) -> TraceEncoder
+): TraceLogger {
+  val encoder = encoderProvider(workflowScope)
+  val processName = name.ifEmpty { workflowType }
+  return encoder.createLogger(
+      processName = processName,
+      threadName = "Profiling"
+  )
+}
+
 /**
  * A [WorkflowDiagnosticListener] that generates a trace file that can be viewed in Chrome by
  * visiting `chrome://tracing`.
- *
- * @param loggerProvider A function that returns a [TraceLogger] that will be used to write trace
- * events. The function gets the [CoroutineScope] that the workflow runtime is running in, as well
- * as the same workflow type description passed to [WorkflowDiagnosticListener.onWorkflowStarted].
  */
 @UseExperimental(VeryExperimentalWorkflow::class)
-class TracingDiagnosticListener(
-  private val memoryStats: MemoryStats = RuntimeMemoryStats,
+class TracingDiagnosticListener internal constructor(
```

Comment:
Wouldn't hurt to put a comment on the internal constructor explaining that it's broken out s.t. we can unit test w/mocked gc collector.

Reply:
Done.